docs/contributing/adding-a-language.md

```diff
@@ -1,95 +1,99 @@
 ---
 slug: adding-a-language
+title: Add support for a new language
+hide_title: true
+description: Add a new language to Semgrep.
+tags:
+  - Contributing to Semgrep
 ---
-How to add support for a new language
-==
-
-This is about adding support for a new programming language in
-semgrep using the
-[tree-sitter](https://tree-sitter.github.io/tree-sitter/)
-technology. While new languages should use tree-sitter,
-semgrep also supports some languages independently if there's a good
-legacy OCaml parser for them. Check for your language in
-[pfff](https://github.com/semgrep/pfff) and if you see it
-in there, [talk to us](/support).
-Otherwise, let's get started.
+
+# How to add support for a new language
+
+import PL from '@site/src/components/Placeholder';
+
+This document is about adding support for a new programming language in Semgrep using the [tree-sitter](https://tree-sitter.github.io/tree-sitter/) technology. Most languages in semgrep use `tree-parser` though you may also need to update a `menhir` parser.
 
 Repositories involved directly:
-* [semgrep](https://github.com/semgrep/semgrep): the semgrep command line program;
-* [ocaml-tree-sitter-semgrep](https://github.com/semgrep/ocaml-tree-sitter-semgrep): language-specific setup, generates C/OCaml parsers for semgrep;
-* A new repo `semgrep-$PL` for the language `$PL`: C/OCaml parser generated from ocaml-tree-sitter-semgrep by an admin.
 
-Submodules overview (semgrep repo)
+* [<code><strong><i class="fas fa-external-link fa-xs"></i> semgrep</strong></code>](https://github.com/semgrep/semgrep): the semgrep command line program.
+* [<code><strong><i class="fas fa-external-link fa-xs"></i> ocaml-tree-sitter-semgrep</strong></code>](https://github.com/semgrep/ocaml-tree-sitter-semgrep): language-specific setup, generates C/OCaml parsers for semgrep.
+* A new repository <code><strong>semgrep-<PL>LANG</PL></strong></code> for the language you're adding: this is a C or OCaml parser generated from `ocaml-tree-sitter-semgrep` by a Semgrep administrator.
+* [<code><strong><i class="fas fa-external-link fa-xs"></i> semgrep-interfaces</strong></code>](https://github.com/semgrep/semgrep-interfaces/blob/main/generate.py)
+
+## Placeholder values
+
+This document uses the placeholder <PL>LANG</PL> to indicate that you should substitute the name of your language as the value in the given context. For example, if your language is Ruby:
```

Comment:
```suggestion
This document uses the placeholder <PL>LANG</PL> to indicate that you should substitute the name of your language as the value in the given context. For example, if your language is Ruby, and the document's instructions read:
```

Reply:
I don't know if this is helpful or not, so feel free to ignore if you don't think it does